### PR TITLE
Update `route_to_<>` types to be strings instead of lists

### PR DESCRIPTION
**Pull request by drewmullen**
This PR cherry-picks from #68 after i reattempted based prior changes (#69).

### CODE REVIEW

File: examples/ipam/main.tf [hunk 1]
```diff
@@ -15,14 +15,14 @@ module "vpc" {
...
-      route_to_nat = true
+      route_to_nat = "0.0.0.0/0"
     }
   }
 }
 
 #####################################
 # Example of a simple IPAM deployment
-# terraform apply -target=module.ipam_base_for_example_only
+# terraform apply -target=module.ipam_base_for_example_only -auto-approve
```
Comment (andrew-glenn) [inline examples/ipam/main.tf:25]: revert this plz

File: main.tf [hunk 2]
```diff
@@ -99,21 +92,29 @@ resource "aws_internet_gateway" "main" {
...
 resource "aws_route" "public_to_tgw" {
-  for_each = length(try(var.subnets.public.route_to_transit_gateway, [])) > 0 ? toset([
-    for _, key in keys(local.subnet_keys.public) : "${key}:${var.subnets.public.route_to_transit_gateway[0]}"
-  ]) : toset([])
+  for_each = (contains(local.subnet_keys, "public") && can(var.subnets.public.route_to_transit_gateway)) ? toset(local.azs) : toset([])
 
-  route_table_id         = awscc_ec2_route_table.public[split(":", each.key)[0]].id
-  destination_cidr_block = var.subnets.public.route_to_transit_gateway[0]
-  transit_gateway_id     = var.subnets.transit_gateway.transit_gateway_id
+  destination_cidr_block     = can(regex("^pl-", var.subnets.public.route_to_transit_gateway)) ? null : var.subnets.public.route_to_transit_gateway
```
Comment (andrew-glenn) [inline main.tf:113]: Why a prefix list here?

**Review by andrew-glenn — CHANGES_REQUESTED** [review]
Nitpicks

**Review by andrew-glenn — CHANGES_REQUESTED** [review]
See previous dismissed review

File: main.tf (hunk 2 above)
Comment (drewmullen) [reply main.tf:113]: users can either specify a singular cidr or a prefix list

File: examples/ipam/main.tf (hunk 1 above)
Comment (drewmullen) [reply examples/ipam/main.tf:25]: done, although this is just a helper command if someone wants to run the test from our example but nbd

**Review by andrew-glenn — APPROVED** [review]

**Pull request merged**